piIOComm.c

```diff
@@ -311,7 +311,7 @@ INT32S piIoComm_sendRS485Tel(INT16U i16uCmd_p, INT8U i8uAddress_p,
 	ret = pibridge_req_gate_tmt(i8uAddress_p, i16uCmd_p, pi8uSendData_p,
 				    i8uSendDataLen_p, pi8uRecvData_p,
 				    rcvlen, timeout);
-	if (ret) {
+	if (ret < 0) {
```

Comment:
I think you need to check for `ret != rcvlen` instead. Otherwise you'll happily accept telegrams which are shorter than expected.